.github/workflows/check-new-releases.yaml

```diff
@@ -0,0 +1,335 @@
+          echo "VERSION=$version" >> $GITHUB_ENV
+          echo "Tarball filename: ${{ env.TARBALL_FILENAME}}"
+          echo "Software version: ${{ env.VERSION }}"
+      
+      - name: Create credentials file
+        env:
+          CONSUMER_KEY: ${{ secrets[inputs.lp-consumer-key] }}
+          ACCESS_TOKEN: ${{ secrets[inputs.lp-access-token] }}
+          ACCESS_SECRET: ${{ secrets[inputs.lp-access-secret] }}
+        run: |
+          cat <<EOF >> ${{ env.LP_CREDENTIALS }}
+          [1]
+          consumer_key = ${CONSUMER_KEY}
+          consumer_secret = 
+          access_token = ${ACCESS_TOKEN}
+          access_secret = ${ACCESS_SECRET}
+          EOF
+
+      - name: Install Python dependencies
+        run: pip install -r requirements.txt
+
+      - name: LP release of packages
+        id: create-new-lp-release
+        run: |
+          echo "Tarball name: ${{ env.TARBALL_FILENAME }} | version: ${{ env.VERSION }}"
+          if [[ "${{ inputs.lp-repo }}" != *"/soss/"* ]]; then
+            python3 launchpad_release.py \
+              --app "${{ inputs.name }}" \
+              --project "${{ inputs.lp-project }}" \
+              --version "${{ env.VERSION }}" \
+              --tarball "${{ env.OUTPUT_DIR }}/${{ matrix.version }}/${{ env.TARBALL_FILENAME }}" \
+              --credentials ${{ env.LP_CREDENTIALS }}
+          else
+            echo "Do not upload any artifacts for repo: ${{ inputs.lp-repo }}"
+          fi
+      - name: Delete previous release
+        uses: cb80/delrel@latest
+        with:
+          tag: ${{ inputs.name }}-${{ env.VERSION }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Release the package
+        id: create-new-release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          release_name: Release ${{ inputs.name }} ${{ env.VERSION }}
+          tag_name: ${{ inputs.name }}-${{ env.VERSION }}
+
+      - name: Upload release asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create-new-release.outputs.upload_url }}
+          asset_path: ${{ env.OUTPUT_DIR }}/${{ matrix.version }}/${{ env.TARBALL_FILENAME }}
+          asset_name: ${{ env.TARBALL_FILENAME }}
+          asset_content_type: application/zip
+
+      - name: Upload checksum
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create-new-release.outputs.upload_url }}
+          asset_path: output/${{ matrix.version }}/${{ env.TARBALL_FILENAME }}.sha512
+          asset_name: ${{ env.TARBALL_FILENAME }}.sha512
+          asset_content_type: application/txt
+
+      - name: Upload Java dependencies to artifactory
+        run: |
+          # unpack tarball content in a temporary directory
```

Comment:
**todo, out-of-scope** Similarly to another comment, I believe that  this should be rather made a python script to make it more maintainable